Source/WebKit/UIProcess/API/glib/WebKitDownloadClient.cpp

```diff
@@ -103,8 +103,12 @@ class DownloadClient final : public API::DownloadClient {
 
     void didFinish(DownloadProxy& downloadProxy) override
     {
-        if (webkitDownloadIsCancelled(m_download.get()))
+        if (webkitDownloadIsCancelled(m_download.get())) {
+#if PLATFORM(GTK) || PLATFORM(WPE)
+            downloadProxy.setDownloadIsCancelled(true);
+#endif
             return;
+        }
```

Comment:
Is this change needed? I think if webkitDownloadIsCancelled returns true here, then DownloadProxy::cancel should have already been called, and there you already call setDownloadIsCancelled.